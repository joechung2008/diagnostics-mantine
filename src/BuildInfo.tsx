import { Table } from "@mantine/core";

const BuildInfo: React.FC<BuildInfoProps> = ({ buildVersion }) => {
  const items = [
    {
      name: "Build Version",
      value: buildVersion,
    },
  ];

  return (
    <Table>
      <Table.Thead>
        <Table.Tr>
          <Table.Th>Name</Table.Th>
          <Table.Th>Value</Table.Th>
        </Table.Tr>
      </Table.Thead>
      <Table.Tbody>
        {items.map((item) => (
          <Table.Tr key={item.name}>
            <Table.Td>{item.name}</Table.Td>
            <Table.Td>{item.value}</Table.Td>
          </Table.Tr>
        ))}
      </Table.Tbody>
    </Table>
  );
};

export default BuildInfo;
