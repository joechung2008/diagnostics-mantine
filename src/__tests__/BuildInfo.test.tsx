import { describe, expect, it } from "vitest";
import BuildInfo from "../BuildInfo";
import { render } from "./test-utils";

describe("BuildInfo", () => {
  it("should render the build version in a table", () => {
    const { container } = render(<BuildInfo buildVersion="1.0.0" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
