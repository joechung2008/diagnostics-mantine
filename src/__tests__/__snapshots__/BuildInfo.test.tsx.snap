// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BuildInfo > should render the build version in a table 1`] = `
<style
  data-mantine-styles="true"
>
  




</style>
`;
